picoCTF-shell/hacksport/deploy.py

```diff
@@ -70,7 +75,8 @@ def give_port():
             "All usable ports are taken. Cannot deploy any more instances.")
 
     while True:
-        port = port_random.randint(0, 65535)
+        # Get a random port that is not already in use.
+        port = port_random.choice([i for i in range(0, 65535) if i not in used_ports])
```

Comment:
> One thing to note. If config is None, then the random int is from 1000-65000. However later on in the same method (give_port()), the port is chosen from 0 - 65535. I'm not sure if this should both be the same or not.

Looking at it, `banned_ports` already covers 0-1000 actually. Might as well go with 1000-65000 then.

Reply:
So I think we probably want to ban up to 1024...

https://www.w3.org/Daemon/User/Installation/PrivilegedPorts.html